### ISSUE TITLE
Referrals not created in MemberMouse when first affiliate deleted in AffiliateWP

### ISSUE THREAD

**Issue by sumobi** [AffiliateWP/AffiliateWP#912]
Ref: https://secure.helpscout.net/conversation/129712202/18780/

A customer's referrals were not being generated in MemberMouse despite everything being set up correctly. I noticed that while testing an affiliate's referral ID of `2`, or even `3`, the ID shown by MemberMouse was always `1`, when you hovered over the affiliate icon, in the MM transaction:

![screen shot 2015-10-15 at 8 50 25 pm](https://cloud.githubusercontent.com/assets/52581/10529729/05f96034-73fd-11e5-8ef5-4342e513dc2b.png)

On the customer's affiliates page, they did not have an affiliate with an ID of `1`. They had added an affiliate, but later deleted the affiliate. After asking the customer to delete all data and re-install (they had only just started luckily) referrals then worked fine.

It seems that this affiliate deletion caused the issue with MM, and it always logged the affiliate as `1`, regardless of the affiliate ID used. And since it always assumed the affiliate was 1 (which didn't exist), no referral was created in AffiliateWP.

**Comment by pippinsplugins** [comment]
.........................

**Comment by pippinsplugins** [comment]
The affiliate ID that is used is actually set by MemberMouse and we just assume that is the ID to use:
```php
$this->affiliate_id = $affiliate_data['order_affiliate_id'];
```
That makes me think this is entirely a MemberMouse bug.

**Comment by DrewAPicture** [comment]
I'm unable to reproduce this issue on the [latest version](http://support.membermouse.com/support/solutions/articles/9000020462-membermouse-versions) of MemberMouse, 2.2.5. I checked the release notes for the last couple of versions and didn't see anything that directly addressed what's described here.

Side note: I did notice that if you set the Affiliate ID setting under the Commission section for a specific MM product, that *will* override any affiliate ID passed via a referral URL, e.g. http://cl.ly/3t2M3S1y163y . It might be worth checking with the customer on whether they were in-effect hard-coding an affiliate ID at the product level.
